### PR TITLE
[DIP] Develop support for Fast Fourier Transform

### PR DESCRIPTION
**Pull request by meshtag**
This PR contains a working 2D FFT implementation for convolutions via frequency domain.

### CODE REVIEW

File: examples/DIPDialect/correlationFFT2D.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
+//===- correlation2D.cpp - Example of buddy-opt tool ----------------------===//
```
Comment (zhanghb97) [inline examples/DIPDialect/correlationFFT2D.cpp:1]: Wrong title here (missing FFT2D).

File: examples/DIPDialect/correlationFFT2D.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements a 2D correlation example with dip.corr_2d operation.
```
Comment (zhanghb97) [inline examples/DIPDialect/correlationFFT2D.cpp:17]: Wrong description here (missing FFT2D).

**Review by zhanghb97 — COMMENTED** [review]
LGTM, but please fix some small typos.
I hope we can add the DIP cases to the integration test (just like the integration test for containers) for the next step.

**Comment by meshtag** [comment]
> I hope we can add the DIP cases to the integration test (just like the integration test for containers) for the next step.

Sure, this seems like a good idea!

File: examples/DIPDialect/correlationFFT2D.cpp [hunk 3]
```diff
@@ -0,0 +1,139 @@
+//===- correlationFFT2D.cpp - Example of buddy-opt tool ----------------------===//
```
Comment (zhanghb97) [inline examples/DIPDialect/correlationFFT2D.cpp:1]: This line exceeds the 80-column limitation.

**Review by zhanghb97 — APPROVED** [review]
Please fix the small code style issue and push the PR directly. Thanks!

**Pull request merged**